diff --git a/LASlib/src/CMakeLists.txt b/LASlib/src/CMakeLists.txt
index b5ad81f..3d7f36e 100644
--- a/LASlib/src/CMakeLists.txt
+++ b/LASlib/src/CMakeLists.txt
@@ -71,7 +71,7 @@ file(GLOB_RECURSE LAS_INCLUDES
 
 add_library(LASlib ${LAS_SRC} ${LAZ_SRC_FULL})
 set_property(TARGET LASlib PROPERTY POSITION_INDEPENDENT_CODE ON)
-set_property(TARGET LASlib PROPERTY CXX_STANDARD 14)
+set_property(TARGET LASlib PROPERTY CXX_STANDARD 17)
 
 if (BUILD_SHARED_LIBS)
 	target_compile_definitions(LASlib PRIVATE "COMPILE_AS_DLL")
